### PR TITLE
feature: Support TF2.12 SageMaker DLC

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

*Description of changes:*

support TF2.12 SageMaker DLC
Add p2 deprecation notice

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/training_compiler/test_tensorflow_compiler.py [hunk 1]
```diff
@@ -57,8 +57,8 @@
 
 @pytest.fixture(scope="module", autouse=True)
 def skip_if_incompatible(tensorflow_training_version, request):
-    if version.parse(tensorflow_training_version) < version.parse("2.9"):
-        pytest.skip("Training Compiler only supports TF >= 2.9")
+    if version.parse("2.9") <= version.parse(tensorflow_training_version) < version.parse("2.12"):
```
Comment (Lokiiiiii) [inline tests/unit/sagemaker/training_compiler/test_tensorflow_compiler.py:60]: Logic is inverted.

Comment (tejaschumbalkar) [reply tests/unit/sagemaker/training_compiler/test_tensorflow_compiler.py]: addressed

**Comment by tejaschumbalkar** [comment]
@Lokiiiiii Ready for review

**Review by Lokiiiiii — APPROVED** [review]

File: src/sagemaker/image_uri_config/tensorflow.json [hunk 2]
```diff
@@ -3755,6 +3756,37 @@
...
+                    "ap-northeast-1": "763104351884",
+                    "ap-northeast-2": "763104351884",
+                    "ap-northeast-3": "364406365360",
+                    "ap-south-1": "763104351884",
+                    "ap-southeast-1": "763104351884",
+                    "ap-southeast-2": "763104351884",
+                    "ap-southeast-3": "907027046896",
+                    "ap-southeast-4": "457447274322",
+                    "ca-central-1": "763104351884",
+                    "eu-central-1": "763104351884",
```
Comment (ShiboXing) [inline src/sagemaker/image_uri_config/tensorflow.json:3776]: cn regions are to be added after the SM platform update right?

**Review by ShiboXing — APPROVED** [review]

**Comment by ShiboXing** [comment]
P2 deprecation changes look good

**Review by vikas0203 — COMMENTED** [review]
/bot run slow-tests,pr

**Review by knikure — COMMENTED** [review]
/bot run all

**Review by knikure — APPROVED** [review]

**Pull request merged**